### PR TITLE
Gem support

### PR DESCRIPTION
**Pull request by atward**
Reorganises things and provides gem support under name 'rspec-teamcity'.

I'd like to suggest changing the github repo name to 'rspec-teamcity' to match the directory structure and rubygem guide. Update spec.homepage in gemspec to match if you do. Includes are now:

```
require 'rspec/teamcity'
```

You could also provide a github release of the gem

```
gem build rspec-teamcity.gemspec
```

### CODE REVIEW

File: custom_formatter.rb [hunk 1]
```diff
@@ -1,11 +0,0 @@
-require "rspec/core/formatters/base_text_formatter"
```
Comment (mike-zorn) [inline custom_formatter.rb:1]: Why did this file get deleted?

File: README.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
-teamcity-rspec
+rspec-teamcity
 =======================
 
-TeamCity's rspec reporter
+TeamCity rspec reporter
 
 TeamCity has a built in rspec reporter, but it only works in certain scenarios.
 I ran into a scenario where I needed to run spec on an EC2 machine and TeamCity
```
Comment (mike-zorn) [inline README.md:7]: Can you add doc for installing via gems?

File: rspec-teamcity.gemspec [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  spec.name          = "rspec-teamcity"
+  spec.version       = Rspec::Teamcity::VERSION
+  spec.authors       = ["Mike Atkins"]
+  spec.email         = ["apeherder@gmail.com"]
+  spec.summary       = %q{TeamCity rspec reporter}
+  spec.description   = <<-EOF
+    Provides an rspec reporter that produces teamcity parseable strings. See TC docs:
+    https://confluence.jetbrains.com/display/TCD8/Build+Script+Interaction+with+TeamCity#BuildScriptInteractionwithTeamCity-ReportingTests
+    EOF
+  spec.homepage      = "https://github.com/apechimp/teamcity-rspec"
```
Comment (mike-zorn) [inline rspec-teamcity.gemspec]: I'll rename the repo, so go ahead and change this.

File: custom_formatter.rb (hunk 1 above)
Comment (atward) [inline custom_formatter.rb:1]: This looked like a PoC/example and isn't part of the rspec-teamcity functionality. It looks like a modified example from https://www.relishapp.com/rspec/rspec-core/v/3-3/docs/formatters/custom-formatters
What functionality does this file provide?

Comment (mike-zorn) [inline custom_formatter.rb:1]: Ahh, OK. Makes sense. To be honest, I haven't mucked with this in a while and have forgotten a bit :smile_cat:

**Comment by atward** [comment]
Rebased with requested documentation/reference changes.

**Comment by atward** [comment]
:+1:

**Comment by mike-zorn** [comment]
```
Successfully registered gem: rspec-teamcity (0.0.1)
```

So that means I can remove the `git: ...` part of the install process in the documentation, right? This is actually my first :gem: :smiley_cat:

**Pull request merged**